common/certificate/src/validate_device_id.rs

```diff
@@ -0,0 +1,75 @@
+use crate::CertificateError;
+
+fn is_valid_device_id_char(ch: char) -> bool {
+    matches!(ch, 'A'..='Z'
+    | 'a'..='z'
+    | '0'..='9'
+    | '\''
+    | '='
+    | '('
+    | ')'
+    | '+'
+    | ','
+    | '-'
+    | '.'
+    | ':'
+    | '?')
+}
+
+pub fn is_valid_device_id(id: &str, max_cn_size: usize) -> Result<(), CertificateError> {
+    if id.is_empty() {
+        return Err(CertificateError::EmptyName);
+    } else if id.len() > max_cn_size {
+        return Err(CertificateError::TooLongName {
+            name: id.into(),
+            max_cn_size,
+        });
+    } else if !id.chars().all(is_valid_device_id_char) || id.contains(char::is_control) {
+        return Err(CertificateError::InvalidCharacter { name: id.into() });
+    }
+
+    Ok(())
+}
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn valid_device_id_with_aplhabets() {
+        assert!("ThinEdgeDevice".chars().all(is_valid_device_id_char));
```

Comment:
The test must use the public API, here the function `is_valid_device_id`, and not some internal piece of code.

Reply:
done